### ISSUE TITLE
Update for Godot 4.x

### ISSUE THREAD

**Issue by IvailoBurov** [fian46/addons-btree#65]
Hello,
Do you have any plans to update this wonderful plugin for the new version of Godot?
Yesterday I tried to adapt it for Godot 4.1, but I couldn't get a working version.

**Comment by fian46** [comment]
Hi i haven't trying this plugins for godot 4.1.x since my game is still in the porting process from 3.x to 4.x. yes there is problem with gdscript 1.x to 2.x conversion that requires tons of refactor and internal api changes in godot editor that requires huge refactor in code base. the plugins will stuck with godot 3.x only until either i refactor the plugins or rewrite it completely since godot editor plugins is quite tricky at 4.1.x

**Comment by IvailoBurov** [comment]
Thanks for the quick response.
I hope for success. If needed I can comment some parts of the code that are problematic in version 4.1.
At this point I no longer have any errors in the code with the new version of the script, but I still haven't achieved a working solution

**Comment by fian46** [comment]
yeah i have been trying to port it for 4.1.x but there is too many API changes inside the editor it self probably it will take weeks to re-implement current feature. sadly it is probably will stay on 3.x.x only. the changes in internal API for window is way too many. pull request are welcome for this.

**Comment by IvailoBurov** [comment]
Thanks for the reply. I will review it next week. As far as I remember I had made an option for advanced text input and I will post that project here as well to save your time. I have no github skills at this stage unfortunately.

**Comment by IvailoBurov** [comment]
Indeed, it takes even more than a week to process. I just finished my translation experiment to Godot 4.2.1 and everything seems to work, including the runtime debugger. Since WebSocketServer does not exist in Godot 4.x, I used a plugin from the asset library. If anyone is interested, I can upload the working translated version here for testing, I just need to clean up some unnecessary comments and print lines I wrote during the translation.

**Comment by fian46** [comment]
Great you can do pull request for it. Or create new branch 4.1 so old version is still supported.
